tests/helper/helper_generic.go

```diff
@@ -177,6 +179,63 @@ func WatchNonRetCmdStdOut(cmdStr string, timeout time.Duration, check func(outpu
 	}
 }
 
+// RunCmdWithMatchOutputFromBuffer starts the command, and command stdout is attached to buffer.
+// we read data from buffer line by line, and if expected string is matched it returns true
+// It is different from WaitforCmdOut which gives stdout in one go using session.Out.Contents()
+// for commands like odo log -f which streams continuous data and does not terminate by their own
+// we need to read the stream data from buffer.
+func RunCmdWithMatchOutputFromBuffer(timeoutAfter time.Duration, matchString, program string, args ...string) (bool, error) {
+	var buf bytes.Buffer
+
+	command := exec.Command(program, args...)
+	command.Stdout = &buf
+
```

Comment:
I think we should also check the StdErr stream. In case of the timeout, we can print the contents from the error buffer. It will help in debugging in case the command fails to stream.

Reply:
done

Reply:
I think it's better to use a different buffer for the error stream like in case of the watch test.

Reply:
done